### PR TITLE
Template content validation

### LINKED ISSUE #16943: Inconsistent error when creating an empty admin template
### Describe the bug

If you try to create a new admin template with empty content, the template is not created but also no error or warning is displayed either. If you edit an existing admin template by clearing out its content, then the `The content is mandatory.` validation message is displayed, but this does not happen when creating a brand new template. This is certainly a bug, and very confusing behavior.

Additionally, it's not made clear why an empty content is not allowed. I know that using Placements to not even display the shape is a more "correct" approach, but that's a more advanced feature. The validation message should funnel the user towards Placements by mentioning the option and perhaps including a link to the relevant documentation page, instead of just saying that the field is mandatory.

### Orchard Core version

Latest dev branch.

### To Reproduce

1. Go to Admin > Design > Templates.
2. Create a template, e.g. name "Menu" and content "custom template".
3. Click Edit on the template.
4. Clear content and click save.
5. Observe the "The content is mandatory." message is present.
6. Delete the template.
7. Do it again, this time with empty content during creation.
8. Observe that the validation message is missing. If you didn't pay attention, you may think the page wasn't even submitted.

### Expected behavior

- The empty content validation message should be the same regardless whether we are creating a new template or editing an existing one.
- The message should be something like "The content is mandatory. If you left the content empty because you want to hide the shape, use the {Placements}(relative link to the Placements menu item) instead. See [the docs](https://docs.orchardcore.net/en/main/reference/modules/Placements/) for more about this feature."

### PR DESCRIPTION
**Pull request by sarahelsaig**
Fixes #16943

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Templates/Controllers/TemplateController.cs [hunk 1]
```diff
@@ -395,4 +365,32 @@ private IActionResult RedirectToReturnUrlOrIndex(string returnUrl)
...
+                templatesDocument.Templates.ContainsKey(model.Name))
+            {
+                ModelState.AddModelError(nameof(TemplateViewModel.Name), S["A template with the same name already exists."]);
+            }
+        }
+
+        if (string.IsNullOrWhiteSpace(model.Content))
+        {
+            var placementsLink = Url.ActionLink("Index", "Admin", new { area = "OrchardCore.Placements" });
+            var docsLink = "https://docs.orchardcore.net/en/main/reference/modules/Placements/";
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Templates/Controllers/TemplateController.cs:391]:
```suggestion
            var docsLink = "https://docs.orchardcore.net/en/main/reference/modules/Placements/";

```

File: src/OrchardCore.Modules/OrchardCore.Templates/Controllers/TemplateController.cs [hunk 2]
```diff
@@ -395,4 +365,32 @@ private IActionResult RedirectToReturnUrlOrIndex(string returnUrl)
...
+            if (!model.Name.Equals(sourceName, StringComparison.OrdinalIgnoreCase) &&
+                templatesDocument.Templates.ContainsKey(model.Name))
+            {
+                ModelState.AddModelError(nameof(TemplateViewModel.Name), S["A template with the same name already exists."]);
+            }
+        }
+
+        if (string.IsNullOrWhiteSpace(model.Content))
+        {
+            var placementsLink = Url.ActionLink("Index", "Admin", new { area = "OrchardCore.Placements" });
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Templates/Controllers/TemplateController.cs:390]: Can we rid off the magic string and use the action, controller names from the helpers

**Review by hishamco — CHANGES_REQUESTED** [review]

**Review by hishamco — APPROVED** [review]

**Comment by Piedone** [comment]
Can this be merged, or are you waiting for anything, @hishamco? If it can, please merge.

**Comment by sebastienros** [comment]
Modules should not reference modules. To remove the dependency use a route name. And if the module/feature is not enabled it could create an issue, so maybe check it first. Or just point to the documentation.

**Comment by hishamco** [comment]
Oops, I didn't notice that because the path did not contain `Modules`, let me revise this again and remove the module reference

**Pull request merged**